pkg/option/config.go

```diff
@@ -3072,6 +3075,63 @@ func (c *DaemonConfig) KubeProxyReplacementFullyEnabled() bool {
 		c.EnableSessionAffinity
 }
 
+// StoreInFile stores the configuration in a the given directory under the file
+// name 'daemon-config.json'. If this file already exists, it is renamed to
+// 'daemon-config-1.json', if 'daemon-config-1.json' also exists,
+// 'daemon-config-1.json' is renamed to 'daemon-config-2.json'
+func (c *DaemonConfig) StoreInFile(dir string) error {
+	backupFileNames := []string{
+		"daemon-config.json",
+		"daemon-config-1.json",
+		"daemon-config-2.json",
```

Comment:
Since this is probably the runtime config of the agent, we should name it closer to that. At least `agent-config`